### ISSUE TITLE
Extending the Anki theme ...

### ISSUE THREAD

**Issue by badlydrawnrob** [bikenik/Anki_Templates#1]
Hey @bikenik — first of all, thanks for using my theme! It's nice to see what you're doing with it.

At some point in the next few months I need to get back to my [Anki programming theme](https://github.com/badlydrawnrob/anki), tidy up (or reinvent) the CSS and make the documentation a little clearer.

I wouldn't want to interrupt your hard work, so would be helpful to hear what you have planned and how I can avoid too many breaking changes.

I'm actually learning a language too! (French) I did build some basic language themes based on [Fluent Forever](https://fluent-forever.com/), but they have a [new app coming out soon](https://www.indiegogo.com/projects/fluent-forever-app-think-in-any-new-language-smartphone#/) which I've signed up for, so it's likely I'll ditch Anki for that (it's still great but takes time to set up the language cards)

- I especially like your [multiple choice](https://github.com/bikenik/anki/blob/master/dist/themes/selectable/README.md)
    + Does it work on AnkiDroid / iOS?
- Perhaps it could be helpful to add a section for custom templates?

Thanks,

Rob

**Comment by bikenik** [comment]
Yeah. Thanks for the base of it, Rob. I've planned to change and commit [fork](https://github.com/bikenik/anki) of your theme in the near future.

- No, it's not working on AnkiDroid because on this platform doesn't send a variable from front to the back side of the card. So if I choose an answer in front it does not save the value after the click by the button "Show answer". [It works](https://www.reddit.com/r/Anki/comments/4mhfmm/passing_state_between_fields/) for macOs, iOS(AnkiMobile), Windows, Linux (not tested). Also, I think it would be more convenient for the mobile clients without [jquery UI selectable](https://jqueryui.com/selectable/). Cause it becomes not scrollable. 
-  section for custom templates. I don't know what it has to look like. I can fork it again with any your solution of organizing. 
p.s. Today I knew about [Reverso extensions](http://www.reverso.net/download-browser-extension). I'm planning to integrate it with Anki using [Anki-Connect](https://github.com/FooSoft/anki-connect)
[update]
And perhaps you'd be interested in [this development](https://www.reddit.com/r/Anki/comments/7i0y41/two_maybe_useful_anki_tools_autocloze_and_a/) of Anki theme for French deck. 
I think that this theme with Murphy system for English grammar could be overwritten to the French grammar also.

**Comment by badlydrawnrob** [comment]
Sounds good!

- **Multiple choice** This is the issue I had with javascript code highlighting, it's too resource heavy and slows down card loading time. From what I've read, it's [not currently possible to include external js](https://www.reddit.com/r/Anki/comments/3q0fs8/how_to_load_external_javascript/) for AnkiDroid and iOS, so only seems a good idea when studying with the desktop app.
    + Is your selectable script loading without any noticable slow-down on iOS?
    + You're right about jQuery. I've dabbled with [Elm](http://elm-lang.org/), but I'm not a Javascript dev.
    + Do you think there'd be a way to simplify the js? For instance, [highlight the selected anchor tag](https://css-tricks.com/forums/topic/highlight-selected-navigation-item-anchor-link-same-page/)? You could then determine if it's the correct answer and style with css.
- **Custom templates** Neither do I! I'll have a look at my code again, it's been a while.
    + I have some [basic instructions here](https://github.com/badlydrawnrob/anki/blob/master/dist/themes/assets/css/README.md#changing-the-default-theme) for non-breaking changes.
    + But ... I'd likely be moving to [ECSS](https://github.com/badlydrawnrob/ecss) and [Stylus](https://github.com/badlydrawnrob/cardinal-stylus). This would allow better targeting of components, but means removing Less.
    + Perhaps the best way is to create an issue with a Roadmap of sorts, where you could comment.

Oh, Reverso and Anki-Connect look cool. I guess you'd be auto generating translations and auto-creating cards? Be interested to see that in action.

And thanks for the French deck. I've not heard of the Murphy system, but definitely having multiple hidden words works nicely (and conjugation cards take a lot of time).

I think one thing I've learned with Anki themes is not to try to be too clever. I'm forever forgetting about my thinking process and what I'm supposed to do with some of the fields! I think your gif demos in the documentation make a lot of sense to use.

**Comment by badlydrawnrob** [comment]
@bikenik Or even simpler, pure css [`#anchor` hightlighting](http://jsfiddle.net/davidThomas/NGyVL/)! I'll add suggestions on the correct repo next time, it's a shame Github has no inbox messaging.

**Comment by bikenik** [comment]
**Multiple choices**
> Is your selectable script loading without any noticeable slow-down on iOS?

- Ok. I removed all additional jQuery from "selectable" theme. and made refactor JS. So it's pure JS (ECMAScript 3) about hundred lines (with comments)
Works fine on my iPhone-6. (responsible design for iPad also) 
And now JS looks like [this](https://github.com/bikenik/anki/blob/master/dist/themes/selectable/js/front-selectable.js) for [front template]
and [this](https://github.com/bikenik/anki/blob/master/dist/themes/selectable/js/back-selectable.js) for [back template] (see the comment on top of [back template])
this code should not affect performance

> Do you think there'd be a way to simplify the js? For instance, [highlight the selected anchor tag](https://css-tricks.com/forums/topic/highlight-selected-navigation-item-anchor-link-same-page/)?

It's a good example, but it works with jQuery features and required for testing on all devices for conflict with native jQuery in Anki.
I hope my solution is enough minimalist and simple. I'll update it in future if I work something out.
my regulations for this theme are: 
- the sequential of fields must be random every time
- additional quiz by click on the back side of the card when the answer already exists.
- ✔ Key point notes - well done: displayed when the answer is correct for Back Template
- ✘ Key point notes - bummer: displayed when the answer is wrong for Back Template

[Introduction to Git_(Multiple_choices_test).apkg](https://github.com/bikenik/anki/blob/master/dist/deck/Introduction%20to%20Git_(Multiple_choices_test).apkg)
p.s.  I lent the design of this theme from [teamtreehouse.com](https://teamtreehouse.com)

> Reverso and Anki-Connect look cool

I couldn't do it, cause it's not open source in Safari. But I can always parse my own page of this service and automate adding a card to Anki from there.

**Comment by badlydrawnrob** [comment]
Hi @bikenik 

It seems like you're making great progress and I agree with your choices to convert to vanilla JS, as well as your "regulations" (`shuffle()` will definitely make you have to think about the question).

I've made a start refactoring code myself, but it'll be a while before I can look at this properly.

I've found it's very easy to get overly complex with themes, so I'm trying to strip out as much fluff as possible, so I don't have those WTF moments when revisiting them later!

Looking good! :)
